tests/test_models.py

```diff
@@ -4667,6 +4667,80 @@ def test_create_from_node(self):
         assert_equal(data, draft.registration_metadata)
         assert_equal(proj, draft.branched_from)
 
+class TestPublicFiles(OsfTestCase):
+    def setUp(self):
+        super(TestPublicFiles, self).setUp()
+        self.user = UserFactory()
+        self.auth = Auth(user=self.user)
+        self.project = PublicFilesFactory(creator=self.user)
+        with self.context:
+            handlers.celery_before_request()
+
+    def tearDown(self):
+        super(TestPublicFiles, self).tearDown()
+        self.project.remove()
+
+
+    def test_cannot_delete_public_files_collection(self):
+        assert_raises(NodeStateError, self.project.remove_node, self.auth)
+
+    def test_public_files_is_right_type(self):
+        assert_equal(self.project.is_public_files_collection, True)
+
+
+    def test_no_name_change(self):
+        with assert_raises(NodeStateError):
+            self.project.set_title('Look at me: I\'m the title now',auth=self.auth)
+
+    def test_forking(self):
+        with assert_raises(NodeStateError):
+            fork = self.project.fork_node(auth=self.auth)
+
+    def test_add_remove_permissions(self):
+        unauthorized_user = UserFactory()
+        with assert_raises(NodeStateError):
+            fork = self.project.add_permission(unauthorized_user,'write')
+
+        with assert_raises(NodeStateError):
+            fork = self.project.remove_permission(unauthorized_user,'write')
+
+    def test_cannot_register_public_node(self):
+        with assert_raises(NodeStateError):
+            self.project.register_node(
+                schema=None,
+                auth=self.auth,
+                data=None
+            )
+
+    def test_remove_creator_as_public_collections_owner(self):
+        with assert_raises(NodeStateError):
+            self.project.remove_contributor(self.user,auth=self.auth)
+
+    def test_update_contributor_public_files_collection(self):
+        newman = UserFactory()
+        with assert_raises(NodeStateError):
+            self.project.add_contributor(contributor=newman,permissions='WRITE',auth=Auth(newman))
+
+    def test_changes_privacy_to_public_files_colletion(self):
+        assert_raises(NodeStateError, self.project.set_privacy, 'private', self.auth)
+        assert_equal(self.project.is_public,True)
+
+    def test_citations_for_public_files(self):
+        with assert_raises(NodeStateError):
+            self.project.add_citation(self.auth)
+            self.project.edit_citation(self.auth,{})
+            self.project.remove_citation(self.auth,{})
+
+    def test_user_merge_with_other_public_files_collection(self):
```

Comment:
The only assert the test should need is to check that the files were properly moved over. So an assert that the appended files are in the users public files project should be enough.

And I think John did the behavior that if there were conflicting files the whole process would stop, but I wouldn't really know.